### PR TITLE
Feature/sim 1625/snr refactor

### PR DESCRIPTION
**Pull request by danielsf**
As per Lynne's request, the signal to noise calculating routines now accept only one bandpass at a time.  You can still pass in a numpy array of magnitudes and get an array of uncertainties out.  They just all need to be in the same bandpass.

There is a supporting pull request on sims_catUtils

### CODE REVIEW

File: python/lsst/sims/photUtils/SignalToNoise.py [hunk 1]
```diff
@@ -307,62 +306,40 @@ def calcSNR_m5(magnitudes, bandpasses, m5, photParams, gamma=None):
...
-                           "passed %d gamma parameters" % len(gamma))
+    @param [out] gamma is  the calculated gamma parameter for the
+    bandpass used here (in case the user wants to call this method again).
 
-    if len(m5) != len(bandpasses):
-        raise RuntimeError("Passed %d bandpasses to " % len(bandpasses) + \
-                           " calcSNR_m5; " + \
-                           "passed %d m5 values" % len(m5))
+    Note: You can also pass in a numpy array of magnitudes calculated
+    in the same bandpass and get a numpy array of SNR out
```
Comment (rbiswas4) [inline python/lsst/sims/photUtils/SignalToNoise.py]: Does this mean that all the magnitudes are for the same m5 and gamma (ie. single pointing many sources)

File: python/lsst/sims/photUtils/SignalToNoise.py [hunk 2]
```diff
@@ -371,15 +348,21 @@ def calcMagError_m5(magnitudes, bandpasses, m5, photParams, gamma=None):
...
     arXiv:0805.2366.  If not provided, this method will calculate it.
 
-    @param [out] is a numpy array of errors in magnitude
+    @param [out] the error associated with the magnitude
+
+    @param [out] gamma is  the calculated gamma parameter for the
+    bandpass used here (in case the user wants to call this method again).
+
+    Note: you can also pass in a numpy of arrays calculated in
+    the same Bandpass and get anumpy of errors out
```
Comment (rbiswas4) [inline python/lsst/sims/photUtils/SignalToNoise.py]: pass in a numpy array of magnitudes (maybe?)

**Comment by rbiswas4** [comment]
Simple questions on docstrings in line comments. Checked tests, good to merge.

File: python/lsst/sims/photUtils/SignalToNoise.py (hunk 1 above)
Comment (danielsf) [inline python/lsst/sims/photUtils/SignalToNoise.py]: Yes.  I will make that more explicit in the docstring.

File: python/lsst/sims/photUtils/SignalToNoise.py (hunk 2 above)
Comment (danielsf) [inline python/lsst/sims/photUtils/SignalToNoise.py]: Yeah.  Thanks.  Guess I was a little hasty trying to get this out.

**Pull request merged**